vividus-to-zephyr-exporter/src/main/java/org/vividus/zephyr/facade/ZephyrFacade.java

```diff
@@ -0,0 +1,145 @@
+
+    private final JiraFacade jiraFacade;
+    private final JiraClient client;
+    private final ZephyrExporterConfiguration zephyrExporterConfiguration;
+
+    public ZephyrFacade(JiraFacade jiraFacade, JiraClient jiraClient,
+            ZephyrExporterConfiguration zephyrExporterConfiguration)
+    {
+        this.jiraFacade = jiraFacade;
+        this.client = jiraClient;
+        this.zephyrExporterConfiguration = zephyrExporterConfiguration;
+    }
+
+    @Override
+    public Integer createExecution(String execution) throws IOException
+    {
+        String responseBody = client.executePost(ZAPI_ENDPOINT + "execution/", execution);
+        List<Integer> executionId = JsonPathUtils.getData(responseBody, "$..id");
+        return executionId.get(0);
+    }
+
+    @Override
+    public void updateExecutionStatus(int executionId, String executionBody) throws IOException
+    {
+        client.executePut(String.format(ZAPI_ENDPOINT + "execution/%s/execute", executionId), executionBody);
+    }
+
+    @Override
+    public ZephyrConfiguration prepareConfiguration() throws IOException
+    {
+        ZephyrConfiguration zephyrConfiguration = new ZephyrConfiguration();
+
+        Project project = jiraFacade.getProject(zephyrExporterConfiguration.getProjectKey());
+        String projectId = project.getId();
+        zephyrConfiguration.setProjectId(projectId);
+
+        String versionId = findVersionId(project);
+        zephyrConfiguration.setVersionId(versionId);
+
+        String projectAndVersionUrlQuery = String.format("projectId=%s&versionId=%s", projectId, versionId);
+
+        String cycleId = findCycleId(projectAndVersionUrlQuery);
+        zephyrConfiguration.setCycleId(cycleId);
+
+        String folderId = findFolderId(cycleId, projectAndVersionUrlQuery);
+        zephyrConfiguration.setFolderId(folderId);
+
+        Map<TestCaseStatus, Integer> statusIdMap = getExecutionStatuses();
+        zephyrConfiguration.setTestStatusPerZephyrIdMapping(statusIdMap);
+
+        return zephyrConfiguration;
+    }
+
+    private String findVersionId(Project project)
+    {
+        return project.getVersions()
+                .stream()
+                .filter(v -> zephyrExporterConfiguration.getVersionName().equals(v.getName()))
+                .findFirst()
+                .map(Version::getId)
+                .orElseThrow(() -> new IllegalArgumentException(String.format(
+                        "Version with name '%s' does not exist", zephyrExporterConfiguration.getVersionName())));
+    }
+
+    private String findCycleId(String projectAndVersionUrlQuery) throws IOException
+    {
+        String json = client.executeGet(ZAPI_ENDPOINT + "cycle?" + projectAndVersionUrlQuery);
+        Map<String, Map<String, String>> cycles = JsonPathUtils.getData(json, "$");
+        cycles.remove("recordsCount");
+        Iterator<Map.Entry<String, Map<String, String>>> itr = cycles.entrySet().iterator();
```

Comment:
It is doable via jsonpath?

Reply:
I can't do it with json path :(
https://github.com/vividus-framework/vividus/pull/1109#discussion_r521923117